plugins/obs-filters/color-grade-filter.c

```diff
@@ -183,6 +189,21 @@ static void *load_cube_file(const char *const path, uint32_t *const width,
 			    struct vec3 *domain_min, struct vec3 *domain_max,
 			    enum clut_dimension *dim)
 {
+#ifndef _WIN32
+	locale_t locale = newlocale(LC_ALL_MASK, "C", 0);
+	locale_t previous = 0;
+	if (locale) {
+		previous = uselocale(locale);
+		if (!previous) {
+			blog(LOG_WARNING, "uselocale: Failed to use C locale");
+			freelocale(locale);
+			locale = 0;
```

Comment:
freelocale could be done in the last block only conditioned on if(locale) if we remove this early free?

Reply:
Updated.

Reply:
The else around the free in the last block can be removed too, since we dont do the early free.

Reply:
I don't think it can be removed. If we can't dislodge the locale, we would be freeing an active locale.

Reply:
Hmm looks like you are right, I see why you did the early free now.

Reply:
The early free was just some conservative code I wrote early on. It's not necessary.

I don't like writing code that can leak, but it's the lesser of evils, and also very unlikely to happen.